homework/tests/example.spec.js

```diff
@@ -17,15 +17,115 @@ const data = [
 
 data.forEach(version => {
   test.describe(version + ': Add', () => {
-    test('Concatenating 2 and 3 results in 23', async ({ page }) => {
+    test('Adding 3,3 and 4 results in 7', async ({ page }) => {
       await page.goto('https://testsheepnz.github.io/BasicCalculator');
       await page.selectOption('#selectBuild', { label: version});
-      await page.locator('#number1Field').type('2');
-      await page.locator('#number2Field').type('3');
+      await page.locator('#number1Field').type('3,3');
+      await page.locator('#number2Field').type('4');
       await page.selectOption('#selectOperationDropdown', {label: 'Add'});
+      await page.locator('#integerSelect').click();
       await page.locator('#calculateButton').click();
   
-      await expect(page.locator('#numberAnswerField')).toHaveValue('5');
+      await expect(page.locator('#numberAnswerField')).toHaveValue('7');
+    });
+    test('Adding -3 and 4 results in 1', async ({ page }) => {
```

Comment:
You have wrote 5 tests just to test Add function. In the homework description it was described "**Tests should cover main functions, critical flows.**".  In the test report it is clearly visible that only 2,7,9 builds are failing with Add function so maybe only single test is enough to check all the builds? If you believe that doing multiple checks just for Add function adds value, separate function should be created and called with different parameters. Please leave only 2 tests regarding Add function and when you call those tests use function.  I am attaching a not finished code snippet which I hope will help you to figure this out. 

```
    async checkAddFunction(pageUrl, buildVersion, numberOne, numberTwo) {
        await page.goto(pageUrl);
        await page.selectOption('#selectBuild', {label: buildVersion});
        await page.locator('#number1Field').type(numberOne);
        ...
        ...
    }
```

```
 test('Adding -3 and 0 results in -3', async ({page}) => {
await checkAddFunction("https://testsheepnz.github.io/BasicCalculator", version, "3" ...);


            await expect(page.locator('#numberAnswerField')).toHaveValue('-3');
        });
    });
```

Disclaimer: I did not test this code and it might need some improvement, it is just for you to get an idea how it should be done